sql/src/test/java/io/crate/executor/transport/RerouteActionsTest.java

```diff
@@ -175,4 +180,39 @@ public void testMoveShardRequest() throws Exception {
         request.add(command);
         assertEquals(request, actualRequest);
     }
+
+    @Test
+    public void testCancelShardRequest() throws Exception {
+        GenericProperties properties = new GenericProperties();
+        properties.add(new GenericProperty("allow_primary", BooleanLiteral.TRUE_LITERAL));
+        RerouteCancelShardAnalyzedStatement statement = new RerouteCancelShardAnalyzedStatement(
+            TableDefinitions.USER_TABLE_INFO,
+            Collections.EMPTY_LIST,
+            SqlParser.createExpression("0"),
+            SqlParser.createExpression("node1"),
+            properties);
+        ClusterRerouteRequest actualRequest = RerouteActions.prepareRequest(statement, Row.EMPTY);
+
+        ClusterRerouteRequest request = new ClusterRerouteRequest();
+        CancelAllocationCommand command = new CancelAllocationCommand(
+            TableDefinitions.USER_TABLE_INFO.ident().indexName(), 0, "node1", true);
+        request.add(command);
+        assertEquals(request, actualRequest);
+    }
+
+    @Test
+    public void testCancelShardRequestWithInvalidProperty() throws Exception {
```

Comment:
I would move that as an AnalyzerTest, and also add a test with invalid property key.

Reply:
@matriv I cannot do that since `Row` params are needed to convert the Expressions. It's not safe to do the conversion in the analyzer since the Row parameters don't have to exist there. I think @mfussenegger mentioned that this will be a problem for other statements as well when "Parameter Description" will get introduced. Doing the conversion within the "action/task" layer is safe. What would be necessary is that the expression conversion in the analyzer will be marked as "deprecated" to avoid any other wrong implementation in future.

Reply:
@matriv the test with the invalid property key exists already

Reply:
@mikethebeer Sorry, I didn't comment before, but the `allow_primary` property should be validated and converted to a `boolean in `AlterTableRerouteAnalyzer` as we do with all other statements, and not in the Execution phase.

Reply:
@matriv This is not possible due to the condition I described above. The property validation is doing the Expression conversion. Doing this in the Analyzer is too early and there might be no `Row` parameters available.

Reply:
@mikethebeer sorry, yes now I got it... :-)